ktor-server/ktor-server-core/common/src/io/ktor/server/routing/RouteSelector.kt

```diff
@@ -661,3 +663,7 @@ internal fun evaluatePathSegmentParameter(
         segmentIncrement = 1
     )
 }
+
+private fun ContentType.isCompatibleWith(other: ContentType): Boolean =
+    this.contentSubtype == "*" || this.match(other)
```

Comment:
Could you check if it works for an unhappy path? For example, `get("/image")` with `Accept: text/plain` shouldn't find any suitable routes.

Reply:
Sorry, I didn't include all the changes. Please take a look at the latest commit